codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/Rust.kt

```diff
@@ -42,6 +42,19 @@ import java.nio.file.Files.createTempDirectory
 import java.nio.file.Path
 import kotlin.io.path.absolutePathString
 
+object Commands {
+    val CargoEnvDWarnings = mapOf(
+        "RUSTFLAGS" to "-D warnings --cfg aws_sdk_unstable",
+    )
+    val CargoEnvDDeadCode = mapOf(
+        "RUSTFLAGS" to "-A dead_code --cfg aws_sdk_unstable",
+    )
```

Comment:
Rather than allowing dead code here, we should generate the `#[allow(dead_code)]` attribute each place where it's needed.

Reply:
https://github.com/awslabs/smithy-rs/pull/1069

>This PR addresses both issues and stops short of disallowing warnings from Rustdoc, which isn't quite possible until we implement https://github.com/awslabs/smithy-rs/issues/1068.

It seems like this was added to address the issues that can't be fixed right now.
I checked https://github.com/awslabs/smithy-rs/issues/1068 but it's still open.

I'd say we should leave it there.